pkg/operatorsource/kubeclient.go

```diff
@@ -0,0 +1,54 @@
+package operatorsource
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+)
+
+// Kubeclient interface wraps the package level sdk.Create and sdk.Get methods provided by operator-sdk
+//
+// Why is this wrapper interface necessary?
+// Because these package level functions dont provide a way to inject a fake client,
+// it's hard to unit test reconciliation logic.
+// As a workaround, the reconciliation logic, instead of using the package level function directly will use
+// this interface. We can mock this interface and thus write proper unit test(s) for reconciliation logic.
+//
+// operator-sdk team is working on integrating controller-runtime
+// which will make it possible to inject fake clients in the future.
+// See https://github.com/operator-framework/operator-sdk/issues/382 for more information
+type Kubeclient interface {
+	// Create creates the provided object on the server and updates the arg
+	// "object" with the result from the server(UID, resourceVersion, etc).
+	// Returns an error if the object’s TypeMeta(Kind, APIVersion) or ObjectMeta(Name/GenerateName, Namespace) is missing or incorrect.
+	// Can also return an api error from the server
+	// e.g AlreadyExists https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L423
+	Create(object sdk.Object) error
+
+	// Get gets the specified object and unmarshals the retrieved data into the "into" object.
+	// "into" is a Object that must have
```

Comment:
Is `into` an accepted usage in k8s? Or is it something you came up with?

Reply:
Not sure about k8s, but this is used by `operator-sdk`. Also, as I mentioned that the wrapper interface has same exact signature as `operator-sdk` functions. I just copied the `sdk` function signature along with their documentation.